### ISSUE TITLE
Using backspace while prompt_toolkit shows options sends shell into eternal exceptions

### ISSUE THREAD

**Issue by Achimh3011** [xonsh/xonsh#3503]
When I open xonsh, start typing a command (say `ls`), press TAB, select one suggestion and then type backspace, an exception occurs in the event loop that cannot be exited.

## xonfig
```
$ xonfig
+------------------+---------------------+
| xonsh            | 0.9.14              |
| Python           | 3.8.2               |
| PLY              | 3.11                |
| have readline    | True                |
| prompt toolkit   | 2.0.10              |
| shell type       | prompt_toolkit2     |
| pygments         | 2.4.2               |
| on posix         | True                |
| on linux         | True                |
| distro           | opensuse-tumbleweed |
| on darwin        | False               |
| on windows       | False               |
| on cygwin        | False               |
| on msys2         | False               |
| is superuser     | False               |
| default encoding | utf-8               |
| xonsh encoding   | utf-8               |
| encoding errors  | surrogateescape     |
+------------------+---------------------+
```

</details>

## Expected Behavior
I open xonsh, start typing a command (say `ls`), press TAB, select one suggestion and then type backspace, an exception occurs in the event loop that cannot be exited.

### Traceback (if applicable)

<details>

```
Unhandled exception in event loop:
  File "/usr/lib/python3.8/site-packages/prompt_toolkit/eventloop/posix.py", line 155, in _run_task
    t()
  File "/usr/lib/python3.8/site-packages/prompt_toolkit/eventloop/context.py", line 116, in new_func
    return func(*a, **kw)
  File "/usr/lib/python3.8/site-packages/prompt_toolkit/application/application.py", line 403, in redraw
    self._redraw()
  File "/usr/lib/python3.8/site-packages/prompt_toolkit/application/application.py", line 471, in _redraw
    self.renderer.render(self, self.layout)
  File "/usr/lib/python3.8/site-packages/prompt_toolkit/renderer.py", line 556, in render
    layout.container.preferred_height(size.columns, size.rows).preferred)
  File "/usr/lib/python3.8/site-packages/prompt_toolkit/layout/containers.py", line 254, in preferred_height
    dimensions = [c.preferred_height(width, max_available_height)
  File "/usr/lib/python3.8/site-packages/prompt_toolkit/layout/containers.py", line 254, in <listcomp>
    dimensions = [c.preferred_height(width, max_available_height)
  File "/usr/lib/python3.8/site-packages/prompt_toolkit/layout/containers.py", line 649, in preferred_height
    return self.content.preferred_height(width, max_available_height)
  File "/usr/lib/python3.8/site-packages/prompt_toolkit/layout/containers.py", line 254, in preferred_height
    dimensions = [c.preferred_height(width, max_available_height)
  File "/usr/lib/python3.8/site-packages/prompt_toolkit/layout/containers.py", line 254, in <listcomp>
    dimensions = [c.preferred_height(width, max_available_height)
  File "/usr/lib/python3.8/site-packages/prompt_toolkit/layout/containers.py", line 2189, in preferred_height
    return self.content.preferred_height(width, max_available_height)
  File "/usr/lib/python3.8/site-packages/prompt_toolkit/layout/containers.py", line 1369, in preferred_height
    return self._merge_dimensions(
  File "/usr/lib/python3.8/site-packages/prompt_toolkit/layout/containers.py", line 1390, in _merge_dimensions
    preferred = get_preferred()
  File "/usr/lib/python3.8/site-packages/prompt_toolkit/layout/containers.py", line 1365, in preferred_content_height
    return self.content.preferred_height(
  File "/usr/lib/python3.8/site-packages/prompt_toolkit/layout/controls.py", line 558, in preferred_height
    content = self.create_content(width, None)
  File "/usr/lib/python3.8/site-packages/prompt_toolkit/layout/controls.py", line 699, in create_content
    cursor_position=translate_rowcol(document.cursor_position_row,
  File "/usr/lib/python3.8/site-packages/prompt_toolkit/layout/controls.py", line 682, in translate_rowcol
    return Point(x=get_processed_line(row).source_to_display(col), y=row)
  File "/usr/lib/python3.8/site-packages/prompt_toolkit/layout/controls.py", line 637, in get_processed_line
    processed_line = transform(i, get_line(i))
  File "/usr/lib/python3.8/site-packages/prompt_toolkit/lexers/pygments.py", line 288, in get_line
    for num, line in generator:
  File "/usr/lib/python3.8/site-packages/prompt_toolkit/formatted_text/utils.py", line 63, in split_lines
    for item in fragments:
  File "/usr/lib/python3.8/site-packages/prompt_toolkit/lexers/pygments.py", line 237, in get_text_fragments
    yield _token_cache[t], v
  File "/usr/lib/python3.8/site-packages/prompt_toolkit/lexers/pygments.py", line 123, in __missing__
    result = 'class:' + pygments_token_to_classname(key)
  File "/usr/lib/python3.8/site-packages/prompt_toolkit/styles/pygments.py", line 61, in pygments_token_to_classname
    parts = ('pygments', ) + token

Exception can only concatenate tuple (not "NoneType") to tuple
Press ENTER to continue... 

Unhandled exception in event loop:
  File "/usr/lib/python3.8/site-packages/prompt_toolkit/eventloop/posix.py", line 155, in _run_task
    t()
  File "/usr/lib/python3.8/site-packages/prompt_toolkit/eventloop/context.py", line 116, in new_func
    return func(*a, **kw)
  File "/usr/lib/python3.8/site-packages/prompt_toolkit/application/application.py", line 403, in redraw
    self._redraw()
  File "/usr/lib/python3.8/site-packages/prompt_toolkit/application/application.py", line 471, in _redraw
    self.renderer.render(self, self.layout)
  File "/usr/lib/python3.8/site-packages/prompt_toolkit/renderer.py", line 556, in render
    layout.container.preferred_height(size.columns, size.rows).preferred)
  File "/usr/lib/python3.8/site-packages/prompt_toolkit/layout/containers.py", line 254, in preferred_height
    dimensions = [c.preferred_height(width, max_available_height)
  File "/usr/lib/python3.8/site-packages/prompt_toolkit/layout/containers.py", line 254, in <listcomp>
    dimensions = [c.preferred_height(width, max_available_height)
  File "/usr/lib/python3.8/site-packages/prompt_toolkit/layout/containers.py", line 649, in preferred_height
    return self.content.preferred_height(width, max_available_height)
  File "/usr/lib/python3.8/site-packages/prompt_toolkit/layout/containers.py", line 254, in preferred_height
    dimensions = [c.preferred_height(width, max_available_height)
  File "/usr/lib/python3.8/site-packages/prompt_toolkit/layout/containers.py", line 254, in <listcomp>
    dimensions = [c.preferred_height(width, max_available_height)
  File "/usr/lib/python3.8/site-packages/prompt_toolkit/layout/containers.py", line 2189, in preferred_height
    return self.content.preferred_height(width, max_available_height)
  File "/usr/lib/python3.8/site-packages/prompt_toolkit/layout/containers.py", line 1369, in preferred_height
    return self._merge_dimensions(
  File "/usr/lib/python3.8/site-packages/prompt_toolkit/layout/containers.py", line 1390, in _merge_dimensions
    preferred = get_preferred()
  File "/usr/lib/python3.8/site-packages/prompt_toolkit/layout/containers.py", line 1365, in preferred_content_height
    return self.content.preferred_height(
  File "/usr/lib/python3.8/site-packages/prompt_toolkit/layout/controls.py", line 558, in preferred_height
    content = self.create_content(width, None)
  File "/usr/lib/python3.8/site-packages/prompt_toolkit/layout/controls.py", line 699, in create_content
    cursor_position=translate_rowcol(document.cursor_position_row,
  File "/usr/lib/python3.8/site-packages/prompt_toolkit/layout/controls.py", line 682, in translate_rowcol
    return Point(x=get_processed_line(row).source_to_display(col), y=row)
  File "/usr/lib/python3.8/site-packages/prompt_toolkit/layout/controls.py", line 637, in get_processed_line
    processed_line = transform(i, get_line(i))
  File "/usr/lib/python3.8/site-packages/prompt_toolkit/lexers/pygments.py", line 288, in get_line
    for num, line in generator:
  File "/usr/lib/python3.8/site-packages/prompt_toolkit/formatted_text/utils.py", line 63, in split_lines
    for item in fragments:
  File "/usr/lib/python3.8/site-packages/prompt_toolkit/lexers/pygments.py", line 237, in get_text_fragments
    yield _token_cache[t], v
  File "/usr/lib/python3.8/site-packages/prompt_toolkit/lexers/pygments.py", line 123, in __missing__
    result = 'class:' + pygments_token_to_classname(key)
  File "/usr/lib/python3.8/site-packages/prompt_toolkit/styles/pygments.py", line 61, in pygments_token_to_classname
    parts = ('pygments', ) + token

Exception can only concatenate tuple (not "NoneType") to tuple
Press ENTER to continue... 
```

This goes on until one kills the shell (e.g. by closing the terminal)
</details>

## Steps to Reproduce

* Open xonsh, 
* start typing a command (say `ls`), 
* press TAB, 
 * select one suggestion and 
* then type backspace

**Comment by gforsyth** [comment]
Thanks for reporting @Achimh3011 -- I can't reproduce this on my end.  It looks like the entirety of the traceback is within `prompt_toolkit`, so can you try upgrading (or downgrading) to a different version?

**Comment by laloch** [comment]
Looks like a duplicate of #3474. Fix should be already merged in #3477.
@Achimh3011, can you reproduce the issue on v0.9.15?

**Comment by gforsyth** [comment]
> Looks like a duplicate of #3474. Fix should be already merged in #3477.

whoops, nice catch, @laloch !

**Comment by Achimh3011** [comment]
Cannot reproduce after upgrading to 0.9.15. Suppose the fix for #3477 caught it. Thanks for checking.
